watch/inotify.go

```diff
@@ -134,3 +141,27 @@ func (fw *InotifyFileWatcher) ChangeEvents(t *tomb.Tomb, pos int64) (*FileChange
 
 	return changes, nil
 }
+
+func (fw *InotifyFileWatcher) checkAndNotifyIfModifiedInBetween(changes *FileChanges) (os.FileInfo, bool) {
+	fi, err := os.Stat(fw.Filename)
+	if err != nil {
+		if os.IsNotExist(err) {
+			RemoveWatch(fw.Filename)
+			changes.NotifyDeleted()
+			return nil, true
+		}
+		util.Fatal("Failed to stat file %v: %v", fw.Filename, err)
```

Comment:
It seems in line with the library, but I find it highly concerning if I can make a program crash by concurrently removing read access from the directory containing a watched file :/

Reply:
Makes sense even though it wouldn't apply to our use case. There are even comments like: `// XXX: report this error back to the user` in the code which I highly doubt would ever actually be fixed. In this case I think we should log an error and treat it as a deleted file